.github/workflows/lock.yml

```diff
@@ -0,0 +1,44 @@
+name: 'Lock Inactive Threads'
+
+on:
+  schedule:
+    # midnight UTC, every Wednesday
+    - cron: '0 0 * * 3'
+  # allow manual triggering from GitHub UI
+  workflow_dispatch:
+
+permissions:
+  issues: write
+  pull-requests: write
+
+concurrency:
+  group: lock
+
+jobs:
+  action:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: dessant/lock-threads@v4
+        with:
+          github-token: ${{ github.token }}
+          # after how many days of inactivity should a closed issue/PR be locked?
+          issue-inactive-days: '90'
+          pr-inactive-days: '90'
```

Comment:
Note that we use to close issues after only 60 days of inactivity: #4251.

I think 90 is a good balance of allowing for recent discussions to be revived while also encouraging the creation of new discussions.

Please also note that this about *locking* 90 days **AFTER an issue/PR is closed**, and that we have a separate bot that automatically *closes* issues after 30 days waiting for a response:

https://github.com/microsoft/LightGBM/blob/5c9e61d19084295fe1c75375baeb9242678d92b1/.github/workflows/no-response.yml#L21